src/core/shader.js

```diff
@@ -60,9 +60,7 @@ Shader.prototype = {
     var self = this;
     var variables = {};
     var schema = this.schema;
-    var squemaKeys = Object.keys(schema);
-    squemaKeys.forEach(processSquema);
-    function processSquema (key) {
```

Comment:
indirection != abstraction. in these cases we're not trying to abstract, the functions are part of the code.